levelkdev/swapr-dapp
# feature: add 0x to Eco Router

# Summary

Fixes #[1053](https://github.com/levelkdev/swapr-dapp/issues/1053)

Integrate 0x Project in Eco Router

![Selection_151](https://user-images.githubusercontent.com/100695408/177161046-29ac59cd-f35f-4fef-99ed-c11a0d7a4998.png)

![Selection_148](https://user-images.githubusercontent.com/100695408/177161133-cef56343-0603-4a27-9ede-cfb0c80423ad.png)

# To Test

1. Open the main page, log in to the wallet, and select mainnet (update: polygon is also supported)
2. Fill fields for the trade. For example, 0X is available for ETH, DAI.
  2a. the exact amount I want to trade
  2b. the exact amount I want to receive
3. Verify if 0X is available in the button loading state
4. Verify if 0X is available in the eco router. Check if price, price impact, and fee are displayed correctly based on API
5. Check other chains if they work as before

# Background

Changes are also in SDK, so need to wait for [them](https://github.com/levelkdev/swapr-sdk/pull/128)

EDIT: Merged

Also, waiting for info from 0x about the unit of estimatedPriceImpact in the API which is displayed as priceImpact. If it is % or decimals

EDIT: estimatedPriceImpact unit is %
